pkgs/applications/window-managers/sway/wrapper.nix

```diff
@@ -15,6 +15,8 @@ let
      set -o errexit
      if [ ! "$_SWAY_WRAPPER_ALREADY_EXECUTED" ]; then
        ${extraSessionCommands}
+       export XDG_SESSION_TYPE=wayland
+       export XDG_CURRENT_DESKTOP=sway
```

Comment:
If there's a reason not to set those modifying the default for `extraSessionCommands` would be better (but only works on NixOS, etc.). Or maybe before `extraSessionCommands` so one can use `unset` instead of setting `_SWAY_WRAPPER_ALREADY_EXECUTED` and (re)`exec`.